src/test/kotlin/org/pkl/lsp/GoToDefinitionTest.kt

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.lsp
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.pkl.lsp.ast.*
+
+class GoToDefinitionTest : LSPTestBase() {
+  @Test
+  fun `resolve property name in scope`() {
+    createPklFile(
+      """
+      prop = 42
+
+      myProp = prop<caret>
+    """
+        .trimIndent()
+    )
+    val resolved = goToDefinition().single()
+    assertThat(resolved).isInstanceOf(PklProperty::class.java)
+    resolved as PklProperty
+    assertThat(resolved.name).isEqualTo("prop")
+  }
+
+  @Test
+  fun `resolve property name this`() {
+    createPklFile(
+      """
+      class MyClass {
+        name: String
+      }
+
+      value: MyClass = new {
+        name<caret> = "Bob"
+      }
+    """
+        .trimIndent()
+    )
+    val resolved = goToDefinition().single()
+    assertThat(resolved).isInstanceOf(PklClassProperty::class.java)
+    resolved as PklClassProperty
+    assertThat(resolved.name).isEqualTo("name")
+    assertThat(resolved.parentOfTypes(PklClass::class)!!.name).isEqualTo("MyClass")
+  }
+
+  @Test
+  fun `resolve function name`() {
+    createPklFile(
+      """
+      function foo() = 42
+      
+      result = fo<caret>o()
+    """
+        .trimIndent()
+    )
+    val resolved = goToDefinition().single()
+    assertThat(resolved).isInstanceOf(PklMethodHeader::class.java)
+    resolved as PklMethodHeader
+    assertThat(resolved.identifier!!.text).isEqualTo("foo")
+  }
+
+  @Test
+  fun `resolve function parameter`() {
+    createPklFile(
+      """
+      function foo(myParam: String) =
+        myParam<caret> + 5
+    """
+        .trimIndent()
+    )
+    val resolved = goToDefinition().single()
+    assertThat(resolved).isInstanceOf(PklTypedIdentifier::class.java)
+    resolved as PklTypedIdentifier
+    assertThat(resolved.identifier!!.text).isEqualTo("myParam")
```

Comment:
nit: we could also assert that `resolved.typeAnnotation!!.type!!.render() == "String"`